src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php

```diff
@@ -82,12 +79,64 @@ public function setMarking(object $subject, Marking $marking, array $context = [
             $marking = key($marking);
         }
 
-        $method = 'set'.ucfirst($this->property);
+        ($this->getSetter($subject))($marking, $context);
+    }
+
+    private function getGetter(object $subject): callable
+    {
+        $propertyName = $this->property;
+        $methodName = 'get'.ucfirst($propertyName);
+
+        return match ($this->getters[$subject] ??= $this->getType($subject, $propertyName, $methodName)) {
+            MarkingStoreMethod::METHOD => static fn () => $subject->{$methodName}(),
+            MarkingStoreMethod::PROPERTY => static fn () => $subject->{$propertyName},
+        };
+    }
+
+    private function getSetter(object $subject): callable
+    {
+        $propertyName = $this->property;
+        $methodName = 'set'.ucfirst($propertyName);
+
+        return match ($this->setters[$subject] ??= $this->getType($subject, $propertyName, $methodName)) {
+            MarkingStoreMethod::METHOD => static fn ($marking, $context) => $subject->{$methodName}($marking, $context),
+            MarkingStoreMethod::PROPERTY => static fn ($marking) => $subject->{$propertyName} = $marking,
+        };
+    }
 
+    private function getType(object $subject, string $propertyName, string $method): MarkingStoreMethod
+    {
         if (!method_exists($subject, $method)) {
-            throw new LogicException(sprintf('The method "%s::%s()" does not exist.', get_debug_type($subject), $method));
+            goto property;
         }
 
-        $subject->{$method}($marking, $context);
+        try {
+            $r = new \ReflectionMethod($subject, $method);
+        } catch (\ReflectionException) {
+            property:
+            try {
+                $r = new \ReflectionProperty($subject, $propertyName);
+            } catch (\ReflectionException) {
+                throw new LogicException(sprintf('The public property "%1$s::%2$s" nor the public method "%1$s::%3$s()" exist. At least one must be declared.', get_debug_type($subject), $propertyName, $method));
+            }
+
+            if (!$r->isPublic()) {
+                throw new LogicException(sprintf('The public method "%1$s::%3$s()" must be declared, or the property "%1$s::%2$s" must be public.', get_debug_type($subject), $propertyName, $method));
+            }
+
+            return MarkingStoreMethod::PROPERTY;
+        }
+
+        if (!$r->isPublic()) {
```

Comment:
if the method is not public but the property is, maybe we should use the property instead of throwing?